app/resources/v1/qc_reception_resource.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module V1
+  # QcReceptionResource
+
+  # Steps:
+  # 1. Create QcReception
+  # 2. Create QcResults
+  # 3. Publish qc result messages
+  class QcReceptionResource < JSONAPI::Resource
+    attributes :qc_results_list, :source
+
+    after_create :publish_messages, :create_qc_results!
```

Comment:
Publish will publish the messages, but first they are created in the create_qc_results. The order of execution should be :create_qc_results! and then  :publish_messages.
Is this ```after_create``` not the other way around?

Reply:
The after_create happens in a transaction, and where there are multiple transactional callbacks, the order of execution is reversed.

Reply:
Ah yes I remember reading about this too. Odd behaviour. Maybe add a comment in this code to specific that is what is happening